src/MOL-SLLOD/compute_pressure_mol.cpp

```diff
@@ -92,143 +82,147 @@ ComputePressureMol::ComputePressureMol(LAMMPS *lmp, int narg, char **arg) :
-  if (pairflag && force->pair) nvirial++;
-  /*
-  if (atom->molecular != Atom::ATOMIC) {
-    if (bondflag && force->bond) nvirial++;
-    if (angleflag && force->angle) nvirial++;
-    if (dihedralflag && force->dihedral) nvirial++;
-    if (improperflag && force->improper) nvirial++;
+    auto ph = dynamic_cast<PairHybrid *>( force->pair);
+    ph->no_virial_fdotr_compute = 1;
+  } else {
+    pair_ptr = force->pair;
   }
-  if (fixflag)
-    for (auto &ifix : modify->get_fix_list())
-      if (ifix->thermo_virial) nvirial++;
-  */
-  if (nvirial) {
-    vptr = new double*[nvirial];
-    nvirial = 0;
-    /*
-    if (pairhybridflag && force->pair) {
-      auto ph = dynamic_cast<PairHybrid *>( force->pair);
-      ph->no_virial_fdotr_compute = 1;
-      vptr[nvirial++] = pairhybrid->virial;
+  if (pairflag || pairhybridflag)
+    pair_ptr->add_tally_callback(this);
+
+  // find fix property/mol
+  molprop = dynamic_cast<FixPropertyMol*>(modify->get_fix_by_id(id_molprop));
+  if (molprop == nullptr)
+    error->all(FLERR, "Compute pressure/mol could not find a fix property/mol with id {}", id_molprop);
+  if (igroup != molprop->igroup)
+    error->all(FLERR, "Compute pressure/mol must be defined for the same group as fix nvt/sllod/mol");
+  // Make sure CoM can be computed
+  molprop->request_com();
+
+  // Make sure ghost atoms have image flags.
+  // Needed to map CoM to same image as ghost atoms.
+  if (!comm->ghost_imageflags)
+    error->all(FLERR, "Fix property/mol requires image flags to be communicated. "
+        "Use comm_modify image yes to enable this.");
+
+  // Warn about possible incompatibilities
+  if (comm->me == 0) {
+    // Check for tally callback issues
+    if ((pairflag || pairhybridflag) &&
+        pair_ptr->single_enable == 0 ||
+        pair_ptr->manybody_flag)
+      error->warning(FLERR,"Compute pressure/mol used with incompatible pair style");
+
+    // Check for fixes that contribute to the virial (currently not handled)
+    for (auto &ifix : modify->get_fix_list()) {
+      if (ifix->thermo_virial)
+        error->warning(FLERR,
+            "Compute pressure/mol does not account for fix virial "
+            "contributions. This warning can be safely ignored for fixes that "
+            "only contribute intramolecular forces");
     }
-    */
-    if (pairflag && force->pair) vptr[nvirial++] = force->pair->molecule_virial;
-    /*
-    if (bondflag && force->bond) vptr[nvirial++] = force->bond->virial;
-    if (angleflag && force->angle) vptr[nvirial++] = force->angle->virial;
-    if (dihedralflag && force->dihedral)
-      vptr[nvirial++] = force->dihedral->virial;
-    if (improperflag && force->improper)
-      vptr[nvirial++] = force->improper->virial;
-    if (fixflag)
-    for (auto &ifix : modify->get_fix_list())
-      if (ifix->virial_global_flag && ifix->thermo_virial)
-          vptr[nvirial++] = ifix->virial;
-    */
+
+    // kspace not yet supported
+    if (kspaceflag && force->kspace)
```

Comment:
```suggestion
    if (kspaceflag && force->kspace && comm->me == 0)
```

warn from only proc 0 instead of all procs

Reply:
That check is already inside an `if (comm->me == 0)` block (see line 196)